### ISSUE TITLE
Removing Navigation Block causes console warning re: state 

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#22830]
**Note** this error is probably not specific to the Navigation block.

**Describe the bug**
When removing the Navigation block from a post the following error appears in the devtools console:

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in ForwardRef(ToolbarContainer) (created by ForwardRef(Toolbar))
    in ForwardRef(Toolbar) (created by NavigableToolbar)
    in NavigableToolbar (created by BlockContextualToolbar)
    in div (created by BlockContextualToolbar)
    in BlockContextualToolbar (created by BlockPopover)
    in div (created by Animate)
    in div (created by ForwardRef)
    in ForwardRef (created by Animate)
    in Animate (created by Popover)
    in PopoverDetectOutside (created by WithFocusOutside(PopoverDetectOutside))
    in div (created by WithFocusOutside(PopoverDetectOutside))
    in WithFocusOutside(PopoverDetectOutside) (created by Popover)
    in Fill (created by Fill)
    in Fill (created by Popover)
    in Popover (created by BlockPopover)
```

When this is fixed we may need to remove the following line from the Navigation block e2e tests added in https://github.com/WordPress/gutenberg/pull/18869:

```js
// Required until fix in place for:
// https://github.com/WordPress/gutenberg/issues/22830
// Ideally we'd be more specific about the error message but it
// does not support partial string matches using `expect.stringContaining()`.
expect( console ).toHaveErrored();
```

**To reproduce**
1. On `master` branch on latest WP 5.5 Enable experiments.
2. Create a post.
3. Add a Navigation block.
4. Remove the block using the block toolbar menu or by selecting an hitting the backspace key.
5. See error in console.

**Expected behavior**
There should be no error in the console.

**Screenshots**
<img width="1067" alt="Screenshot 2020-06-02 at 15 12 13" src="https://user-images.githubusercontent.com/444434/83530783-1e88c500-a4e4-11ea-963d-d20bde584842.png">

**Editor version (please complete the following information):**
- WordPress version: [e.g: 5.3.2]: 5.5
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? [e.g: "gutenberg plugin", "default"]: Current `master` branch.
- If the Gutenberg plugin is installed, which version is it? [e.g., 7.6]: Current `master` branch.

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]: MacOS
 - Browser [e.g. chrome, safari]: Chrome
 - Version [e.g. 22]: 83

**Comment by talldan** [comment]
Just linking to my comment over here https://github.com/WordPress/gutenberg/pull/18869#issuecomment-637973207

TLDR—the above warning should be fixed, but `expect( console ).toHaveErrored();` shouldn't be required in the tests.

**Comment by getdave** [comment]
Yeh I'm not actually sure this is anything specific to the Nav Block. Let me take a look at the advice in your comment. Much appreciated.

**Comment by haszari** [comment]
I see this same error in WordPress 5.4.2 when clicking away from the block popover menu.

- Open browser dev tools, show console.
- Add a block to page, or select an existing block.
- Click three dots menu to show block menu (`Hide Block Settigns`, `Remove Block` etc).
- Click another block or click away to dismiss popover menu.

See this React error in console:

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in SlotComponent (created by Context.Consumer)
    in Slot (created by slot_fill_Slot)
    in slot_fill_Slot (created by __experimentalBlockSettingsMenuFirstItemSlot)
    in __experimentalBlockSettingsMenuFirstItemSlot (created by Dropdown)
    in div (created by MenuGroup)
    in div (created by MenuGroup)
    in MenuGroup (created by Dropdown)
    in div (created by NavigableContainer)
    in NavigableContainer (created by ForwardRef(NavigableContainer))
    in ForwardRef(NavigableContainer) (created by ForwardRef(NavigableMenu))
    in ForwardRef(NavigableMenu) (created by Dropdown)
    in div (created by Animate)
    in div (created by ForwardRef)
    in ForwardRef (created by Animate)
    in Animate (created by Popover)
    in PopoverDetectOutside (created by WithFocusOutside(PopoverDetectOutside))
    in div (created by WithFocusOutside(PopoverDetectOutside))
    in WithFocusOutside(PopoverDetectOutside) (created by Popover)
    in Unknown (created by FocusReturn)
    in div (created by FocusReturn)
    in FocusReturn (created by Context.Consumer)
    in WithFocusReturn(Component) (created by WithConstrainedTabbing(WithFocusReturn(Component)))
    in div (created by WithConstrainedTabbing(WithFocusReturn(Component)))
    in WithConstrainedTabbing(WithFocusReturn(Component)) (created by Popover)
    in bubbles_virtually_fill_Fill (created by slot_fill_Fill)
    in slot_fill_Fill (created by Popover)
    in span (created by Popover)
    in Popover (created by Dropdown)
    in div (created by Dropdown)
    in Dropdown (created by DropdownMenu)
    in DropdownMenu (created by BlockActions)
    in div (created by ToolbarGroupContainer)
    in ToolbarGroupContainer (created by ToolbarGroup)
    in ToolbarGroup (created by toolbar_Toolbar)
    in toolbar_Toolbar (created by BlockActions)
    in BlockActions (created by WithDispatch(BlockActions))
    in WithDispatch(BlockActions)
    in Unknown (created by WithSelect(WithDispatch(BlockActions)))
    in WithSelect(WithDispatch(BlockActions)) (created by BlockSettingsMenu)
    in BlockSettingsMenu (created by BlockToolbar)
    in div (created by BlockToolbar)
    in BlockToolbar (created by BlockContextualToolbar)
    in div (created by NavigableContainer)
    in NavigableContainer (created by ForwardRef(NavigableContainer))
    in ForwardRef(NavigableContainer) (created by ForwardRef(NavigableMenu))
    in ForwardRef(NavigableMenu) (created by NavigableToolbar)
    in NavigableToolbar (created by BlockContextualToolbar)
    in BlockContextualToolbar (created by BlockPopover)
    in div (created by Animate)
    in div (created by ForwardRef)
    in ForwardRef (created by Animate)
    in Animate (created by Popover)
    in PopoverDetectOutside (created by WithFocusOutside(PopoverDetectOutside))
    in div (created by WithFocusOutside(PopoverDetectOutside))
    in WithFocusOutside(PopoverDetectOutside) (created by Popover)
    in bubbles_virtually_fill_Fill (created by slot_fill_Fill)
    in slot_fill_Fill (created by Popover)
    in Popover (created by BlockPopover)
```

**Comment by diegohaz** [comment]
@haszari Looks like that one is related to  https://github.com/WordPress/gutenberg/issues/17355#issuecomment-629934483

**Comment by diegohaz** [comment]
I believe those warnings must have gone away in all blocks except Navigation (which #23281 should hopefully fix). Can someone confirm that?

**Comment by adamziel** [comment]
In my testing:

* The warning is there when I go back to the version from Jun 1st.
* The warning isn't there on the latest master branch.

I'm closing this as solved, feel free to revive if the problem shows up again in the recent version of Gutenberg.
